sql/sqlstats/sslocal: TestSQLStatsDiscardStatsOnFingerprintLimit failed

sql/sqlstats/sslocal.TestSQLStatsDiscardStatsOnFingerprintLimit [failed](https://mesolite.cluster.engflow.com/invocations/default/79335558-a2e5-4de4-a8ea-4192b8eb5041?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL3NzbG9jYWw6c3Nsb2NhbF90ZXN0) on release-25.4 @ [0d364f1ff6ba90d046dc20e30cb9650f079164e9](https://github.com/cockroachdb/cockroach/commits/0d364f1ff6ba90d046dc20e30cb9650f079164e9):


```
=== RUN   TestSQLStatsDiscardStatsOnFingerprintLimit
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestSQLStatsDiscardStatsOnFingerprintLimit3870222042
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_stats_test.go:1800: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsDiscardStatsOnFingerprintLimit3870222042
--- FAIL: TestSQLStatsDiscardStatsOnFingerprintLimit (17.92s)
=== RUN   TestSQLStatsDiscardStatsOnFingerprintLimit/statement_and_txn_limit_reached_across_applications
    sql_stats_test.go:1788: condition failed to evaluate within 5s: from testutils.go:512: expected 1 transaction(s), got 0 filters=
--- FAIL: TestSQLStatsDiscardStatsOnFingerprintLimit/statement_and_txn_limit_reached_across_applications (6.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsDiscardStatsOnFingerprintLimit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57592

--- a/pkg/sql/sqlstats/sslocal/BUILD.bazel
+++ b/pkg/sql/sqlstats/sslocal/BUILD.bazel
@@ -74,6 +74,7 @@ go_test(
         "//pkg/sql/sqlstats/ssmemstorage",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util",

--- a/pkg/sql/sqlstats/sslocal/sql_stats_test.go
+++ b/pkg/sql/sqlstats/sslocal/sql_stats_test.go
@@ -35,6 +35,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/ssmemstorage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
@@ -1640,6 +1641,7 @@ FROM crdb_internal.statement_statistics WHERE app_name = $1`, appName)
 func TestSQLStatsDiscardStatsOnFingerprintLimit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderRace(t)
 
 	ctx := context.Background()
 	s := serverutils.StartServerOnly(t, base.TestServerArgs{})

